apps/discord-bot/src/triggers/cta-post.ts

```diff
@@ -0,0 +1,280 @@
+/* eslint-disable no-console */
+import { ChannelType, type PublicThreadChannel, ThreadAutoArchiveDuration, type Message, MessageFlags } from "discord.js";
+import { type Trigger } from "./trigger";
+import { BarController, Colors, BarElement, CategoryScale, Chart, LinearScale, PieController, ArcElement, Legend, Title } from "chart.js";
+import { Canvas } from "canvas";
+import { writeFile } from "node:fs";
+
+const channelName = "call-to-action";
+const regionRoles = [
+  'Midwest Squad',
+  'South Squad',
+  'Northeast Squad',
+  'West Coast Squad'
+]
+const finishedEmoji = '✅'
+const oneHour = 3_600_000
+const d = new Date();
+
+export class CTAPostTrigger implements Trigger {
+  requireGuild: boolean;
+  public triggered(msg: Message): boolean {
+    // eslint-disable-next-line @typescript-eslint/no-shadow
+    const ctaChannel = msg.guild?.channels.cache.find(ctaChannel => ctaChannel?.name === channelName)
+    // check cta channel exists
+    if (ctaChannel !== undefined) {
+
+      // check message came from cta channel
+      if (msg.channelId === ctaChannel.id) {
+        return true
+      }
+    }
+
+    return false
+  }
+
+  public async execute(msg: Message): Promise<void> {
+
+    if (msg === undefined) {
+      return
+    }
+
+    if (msg.channel.type === ChannelType.GuildText) {
+
+      // check if message created in last month
+      if (msg.createdTimestamp >= d.getMonth() - 1) {
+
+        // check if message already has a thread
+        if (msg.hasThread) {
+
+          if (msg.thread?.archived) {
+            msg.thread.setArchived(false, "Updating chart...")
+          }
+          // is it active?
+          const thread = await this.activeCTAThread(msg)
+          if (thread != undefined) {
+
+            // update the chart in the thread
+            this.updateChart(msg, thread)
+          }
+
+          return
+        }
+
+        // create thread and start collector
+        const thread = await this.createCTAThread(msg)
+        if (thread) {
+          this.startCTAReactionCollector(msg, thread)
+        }
+      }
+    }
+  }
+
+  private async getFinishedReactions(msg: Message): Promise<object> {
```

Comment:
We should be able to specify a more specific return type here, to improve maintainability.